src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.WorkItems
+{
+    public static class SarifWorkItemsExtensions
+    {
+        public static bool AppropriateForFiling(this Result result)
+        {
+            // Fail: an explicit failure occurred.
+            //
+            // Open: a sound analysis is 'open' indicating that the analysis requires
+            //       more configuration or other data in order to produce a determination.
+            // 
+            // Review: an open item which can't be automated, i.e., which requires a
+            //         manual review to resolve.
+            return result.Kind == ResultKind.Fail ||
+                   result.Kind == ResultKind.Open ||
+                   result.Kind == ResultKind.Review;
+
+            // Designations which are not appropriate for filing:
+            //
+            // Pass: the scan target explicitly passed analysis.
+            // 
+            // Not applicable: analysis was skipped because it was not
+            //                 relevant to the scan target.
+            //
+            // Informational: a strictly informational message was emitted.
+        }
+
+        public static string CreateWorkItemTitle(this Run run)
+        {
+            Result firstResult = null;
+            var ruleIds = new HashSet<string>();
+
+            foreach (Result result in run?.Results)
```

Comment:
>run? [](start = 38, length = 4)

If run is null, foreach will throw a NullReferenceException

Reply:
Well, yes. And if run.Results is null that will throw as well. :P Fixing, nice catch. I'm surprised LGTM doesn't catch this, could be worth a problem report. @hakohli

---
In reply to: [392462133](https://github.com/microsoft/sarif-sdk/pull/1813#discussion_r392462133) [](ancestors = 392462133)